engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -1141,6 +1141,10 @@ namespace dmGameObject
         return instance;
     }
 
+    static void Unlink(Collection* collection, Instance* instance);
+    static void ReparentChildNodes(Collection* collection, Instance* instance);
+    static void MoveAllUp(Collection* collection, Instance* instance);
```

Comment:
not sure if I should move a lot of code from one place to another. This looks a bit more clear from diff point of view, but not sure it it's fine.

Reply:
Yeah, not sure either. Forward declarations like this are not very common in our codebase.